src/compas/datastructures/mesh/join.py

```diff
@@ -4,57 +4,105 @@
 
 from copy import deepcopy
 
-from compas.utilities import geometric_key
-
-from compas.datastructures.mesh.clean import mesh_delete_duplicate_vertices
+from compas.utilities import pairwise
 
+from compas.utilities import geometric_key
+from compas.utilities import reverse_geometric_key
 
 __all__ = [
+    'mesh_weld',
```

Comment:
is this the opposite of `mesh_unweld_vertices`? should it then not be in `weld.py`